lib/puppet/provider/elb_loadbalancer/v2.rb

```diff
@@ -298,7 +298,7 @@ def update_listeners
     # which to add, and which need modification.  The modification is a bit of
     # a misnomer, since it really ends up being a delete followed by an add.
 
-    is_listener_ports = @property_hash[:listeners].collect {|x| x['load_balancer_port'].to_i }
+    is_listener_ports = Array(@property_hash[:listeners]).collect {|x| x['load_balancer_port'].to_i }
```

Comment:
And even though the `listeners` property is supposed to accept an array according to the type, it accepts an array of hashes according to https://github.com/puppetlabs/puppetlabs-aws#usage but an array of arrays of hashes according to https://github.com/puppetlabs/puppetlabs-aws/blob/master/lib/puppet/type/elb_loadbalancer.rb#L33-L39 . Something is fishy.